src/providers.ts

```diff
@@ -128,9 +132,61 @@ export const sourcehut: TemplateProvider = (input, options) => {
   };
 };
 
+export const git: TemplateProvider = (input) => {
+  const { uri: gitUri, version } = parseGitCloneURI(input);
+
+  const name = gitUri
+    .replace(/^.+@/, "")
+    .replace(/(\.git)?(#.*)?$/, "")
+    .replaceAll(/[:/]/g, "-");
+
+  return {
+    name,
+    version,
+    tar: async () => {
+      // Lazily import simple-git so we can mark the dependency as optional
+      const { simpleGit: gitCmd } = await import("simple-git");
+
+      // Make temp working directory
+      const tempDir = await mkdtemp(join(tmpdir(), "giget-"));
+
+      // If we do not have version, we can speed up via --depth=1.
+      // Otherwise, we need to clone the entire history, then check out the ref.
+      //
+      // NOTE: We can use git ls-remote to find out if it is a branch, instead of
+      // implementing our own custom notation.
+      if (version) {
+        const branch = version.startsWith("!") && version.slice(1);
+        await gitCmd().clone(gitUri, tempDir, {
+          ...(branch && {
+            "--branch": branch,
+            // Need to pass null for simple-git option with no value
+            // eslint-disable-next-line unicorn/no-null
+            "--single-branch": null,
+          }),
+        });
+
+        // If it is not a branch, we need to checkout to the specific version
+        if (!branch) {
+          await gitCmd({ baseDir: tempDir }).checkout(version);
+        }
+      } else {
+        await gitCmd().clone(gitUri, tempDir, { "--depth": 1 });
+      }
+
+      // Remove .git
+      await rm(join(tempDir, ".git"), { force: true, recursive: true });
```

Comment:
Doesn't tar create have any option to ignore some patterns?

Reply:
Yes, checking on the readme I think we can use `filter`:

> To filter entries, add filter: <function> to the options. Tar-creating methods call the filter with filter(path, stat). Tar-reading methods (including extraction) call the filter with filter(path, entry). The filter is called in the this-context of the Pack or Unpack stream object.